TestFoundation/TestNSNull.swift

```diff
@@ -0,0 +1,42 @@
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2015 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+
+
+
+#if DEPLOYMENT_RUNTIME_OBJC || os(Linux)
+    import Foundation
+    import XCTest
+#else
+    import SwiftFoundation
+    import SwiftXCTest
+#endif
+
+
+
+class TestNSNull : XCTestCase {
+    
+    var allTests: [(String, () -> Void)] {
+        return [
+            ("test_alwaysEqual", test_alwaysEqual)
+        ]
+    }
+    
+    func test_alwaysEqual() {
+        let null_1 = NSNull()
+        let null_2 = NSNull()
+        
+        let null_3: NSNull? = NSNull()
+        let null_4: NSNull? = nil
+        
+        XCTAssertEqual(null_1, null_2)
+        XCTAssertEqual(null_1, null_3)
```

Comment:
in this particular case it preserves the effective functioning behavior of the "singleton-ness" of NSNull without needing a factory pattern init method. I think it is in the spirit of the API to preserve this and add an overload of the === operator to satisfy the expectation that one construction of NSNull() is to be treated as exactly the same as another construction of NSNull()